### PR TITLE
feat: add open table interface and add an example to read table

### PR DESCRIPTION
**Pull request by xudong963**

### CODE REVIEW

File: src/table.rs [hunk 1]
```diff
@@ -75,6 +78,33 @@ impl Table {
     }
 }
 
+/// Open an iceberg table by uri
+pub async fn open(uri: impl AsRef<str>) -> Result<Table> {
```
Comment (Xuanwo) [inline src/table.rs:82]: - I prefer to accept `uri: &str` for more simple to under stand API.
- And I prefer to provide API that called `Table::open`.

File: src/table.rs [hunk 2]
```diff
@@ -75,6 +78,33 @@ impl Table {
...
+    // Todo(xudong): inferring storage types by uri
+    let mut builder = Fs::default();
+    builder.root(uri.as_ref());
+
+    let op = Operator::new(builder)?
+        .layer(LoggingLayer::default())
+        .finish();
+
+    let mut table = Table::new(op);
+    table.load().await?;
```
Comment (Xuanwo) [inline src/table.rs:92]: load should be called by user itself so that they can decide when to load or load again.

File: src/table.rs [hunk 3]
```diff
@@ -75,6 +78,33 @@ impl Table {
...
+    Ok(table)
+}
+
+impl Display for Table {
```
Comment (Xuanwo) [inline src/table.rs:96]: I don't think it's a good idea to implement `Display` for Table. Perhaps we should wait until we find a use case for it before adding it?

File: src/table.rs (hunk 2 above)
Comment (xudong963) [reply src/table.rs:92]: For the `open` semantic, I think should call `load` once, or the table doesn't be opened in fact.

File: src/table.rs (hunk 3 above)
Comment (xudong963) [reply src/table.rs:96]: ok

File: src/table.rs (hunk 2 above)
Comment (Xuanwo) [reply src/table.rs]: Makes sense.

**Review by Xuanwo — APPROVED** [review]
Thanks!

**Pull request merged**